### PR TITLE
fix bugs in  truncated_normal; add TruncatedNormal init.

### PR DESCRIPTION
**Pull request by charlielam0615**
## Description
1. `random.truncated_normal` is not properly truncated. 
2. add support for `TruncatedNormal` initialization
bug fix is a re-implementation of PR#574, also fix bugs therein.
Passed local test.

## Types of changes
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Checklist
- [x] Code follows the code style of this project.
- [x] Changes follow the **CONTRIBUTING** guidelines.
- [x] Update necessary documentation accordingly.
- [x] Lint and tests pass locally with the changes.
- [x] Check issues and pull requests first. You don't want to duplicate effort.

## Other information

### CODE REVIEW

File: brainpy/_src/math/random.py [hunk 1]
```diff
@@ -791,24 +791,27 @@ def uniform(self, low=0.0, high=1.0, size=None, key=None):
...
     return _return(r)
 
-  def truncated_normal(self, lower, upper, size=None, scale=None, key=None):
-    lower = _as_jax_array(lower)
-    lower = _check_py_seq(lower)
-    upper = _as_jax_array(upper)
-    upper = _check_py_seq(upper)
-    scale = _as_jax_array(scale)
-    scale = _check_py_seq(scale)
+  def truncated_normal(self, lower, upper, size=None, scale=1, loc=0, key=None):
```
Comment (chaoming0625) [inline brainpy/_src/math/random.py:794]: ``truncated_normal`` sampling has been revised. Please see #574

File: brainpy/_src/initialize/random_inits.py [hunk 2]
```diff
@@ -122,6 +123,50 @@ def __repr__(self):
     return f'{self.__class__.__name__}(scale={self.scale}, rng={self.rng})'
 
 
+class TruncatedNormal(_InterLayerInitializer):
```
Comment (chaoming0625) [inline brainpy/_src/initialize/random_inits.py:126]: Great!

**Review by chaoming0625 — CHANGES_REQUESTED** [review]
I think we should first merge #574, then merge the latest changes, and finnally the ``brainpy.init.TruncatedNormal`` should work. 

Moreover, please add the new APIs into the ``bainpy.intialize`` module and corresponding doc in ``docs\apis\initialize.rst``

**Comment by charlielam0615** [comment]
@chaoming0625 
PR #574 merged.
New API added.

**Review by chaoming0625 — APPROVED** [review]
So great PR!

**Pull request merged**